tools/test-proxy/Azure.Sdk.Tools.TestProxy/Common/SanitizerDictionary.cs

```diff
@@ -43,21 +47,652 @@ public class SanitizerDictionary
+                    new HeaderRegexSanitizer("MS-CV"),
+                    "AZSDK2020"
+                ),
+                new RegisteredSanitizer(
+                    new HeaderRegexSanitizer("X-Azure-Ref"),
+                    "AZSDK2021"
+                ),
+                new RegisteredSanitizer(
+                    new HeaderRegexSanitizer("x-ms-request-id"),
+                    "AZSDK2022"
+                ),
+                new RegisteredSanitizer(
+                    new HeaderRegexSanitizer("x-ms-client-request-id"),
+                    "AZSDK2023"
+                ),
+                new RegisteredSanitizer(
+                    new HeaderRegexSanitizer("x-ms-content-sha256"),
+                    "AZSDK2024"
+                ),
+                new RegisteredSanitizer(
+                    new HeaderRegexSanitizer("Content-Security-Policy-Report-Only"),
+                    "AZSDK2025"
+                ),
+                new RegisteredSanitizer(
+                    new HeaderRegexSanitizer("Repeatability-First-Sent"),
+                    "AZSDK2026"
+                ),
+                new RegisteredSanitizer(
+                    new HeaderRegexSanitizer("Repeatability-Request-ID"),
+                    "AZSDK2027"
+                ),
+                new RegisteredSanitizer(
+                    new HeaderRegexSanitizer("repeatability-request-id"),
+                    "AZSDK2028"
+                ),
+                new RegisteredSanitizer(
+                    new HeaderRegexSanitizer("repeatability-first-sent"),
+                    "AZSDK2029"
+                ),
+                new RegisteredSanitizer(
+                    new HeaderRegexSanitizer("P3P"),
+                    "AZSDK2030"
                 ),
+                new RegisteredSanitizer(
+                    new HeaderRegexSanitizer("x-ms-ests-server"),
+                    "AZSDK2031"
+                ),
+                #endregion
+                #region BodyRegex
+                new RegisteredSanitizer(
+                    new BodyRegexSanitizer(regex: "(client_id=)(?<cid>[^&]+)", groupForReplace: "cid"),
+                    "AZSDK3000"
+                ),
+                new RegisteredSanitizer(
+                    new BodyRegexSanitizer(regex: "client_secret=(?<secret>[^&\\\"]+)", groupForReplace: "secret"),
+                    "AZSDK3001"
+                ),
+                new RegisteredSanitizer(
+                    new BodyRegexSanitizer(regex: "client_assertion=(?<secret>[^&\\\"]+)", groupForReplace: "secret"),
+                    "AZSDK3002"
+                ),
+                new RegisteredSanitizer(
+                    new BodyRegexSanitizer(regex: "(?:(sv|sig|se|srt|ss|sp)=)(?<secret>[^&\\\"\\s]*)", groupForReplace: "secret"),
+                    "AZSDK3003"
+                ),
+                new RegisteredSanitizer(
+                    new BodyRegexSanitizer(regex: "token=(?<token>[^&]+)($|&)", groupForReplace: "token"),
```

Comment:
Noticed that this regex is breaking some tests since double quotes aren't checked. Should we use similar regex to how `client_assertion` and `client_secret` is handled?

For context, I am encountering an issue in some of the keyvault tests where I have a response body that ends in:
```
...
          {
            "kid": "https://vaultname.vault.azure.net/keys/livekvtestkey08d2f37be",
            "attributes": {
              "enabled": true,
              "created": 1667331450,
              "updated": 1667331450,
              "recoveryLevel": "CustomizedRecoverable\u002BPurgeable"
            }
          }
        ],
        "nextLink": "https://vaultname.vault.azure.net:443/keys?api-version=2016-10-01\u0026$skiptoken=abcdefc12323"
      }
```
This gets sanitized to `-....01&$skiptoken=Sanitized`. And the `"}` at the end gets cut off, leading to malformed JSON.

Reply:
Definitely worth the fix on this side. Will work it in.

Reply:
@pvaneck I'm making this change for _all_ of these BodyRegexSanitizers 🦖